In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('telco_customer_churn.csv')

### Welcome to Day 9 Homework. Please answer the questions below!

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [4]:
df_tugas = df.copy()

In [5]:
df_tugas

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

## Soal nomor 1

In [6]:
jawaban = df_tugas[df_tugas['tenure']>0]

In [7]:
jawaban

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7032 rows x 21 columns]

In [8]:
len(jawaban)  # baris setelah dilakukan filtering terhadap tenure TIDAK SAMA DENGAN 0

7032

## Soal nomor 2

In [9]:
df_condition = df_tugas[(df_tugas['Partner'] == 'No') & (df_tugas['Dependents'] == 'No')]

In [10]:
df_condition

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
...          ...     ...            ...     ...        ...     ...   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
1             Yes                No             DSL                  Yes  ...   
2             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
5             Yes               Yes     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
7034          Yes               Yes     Fiber optic                  Yes  ...   
7035          Yes                No     Fiber optic                   No  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
1                     Yes                   No                   No   
2                      No                   No                   No   
3                     Yes                  Yes                   No   
4                      No                   No                   No   
5                     Yes                   No                  Yes   
...                   ...                  ...                  ...   
7034                  Yes                   No                  Yes   
7035                   No                   No                  Yes   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
1                      No        One year               No   
2                      No  Month-to-month              Yes   
3                      No        One year               No   
4                      No  Month-to-month              Yes   
5                     Yes  Month-to-month              Yes   
...                   ...             ...              ...   
7034                   No  Month-to-month              Yes   
7035                   No  Month-to-month              Yes   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
1                  Mailed check          56.95        1889.5    No  
2                  Mailed check          53.85        108.15   Yes  
3     Bank transfer (automatic)          42.30       1840.75    No  
4              Electronic check          70.70        151.65   Yes  
5              Electronic check          99.65         820.5   Yes  
...                         ...            ...           ...   ...  
7034    Credit card (automatic)         102.95       6886.25   Yes  
7035  Bank transfer (automatic)          7

In [11]:
jawaban = df_condition['tenure'].mean()

In [12]:
jawaban

23.303353658536587

## Soal nomor 3

In [13]:
df_condition2 = df_tugas[(df_tugas['Partner'] == 'Yes') & (df_tugas['Dependents'] == 'No')]

In [14]:
df_condition2

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
8     7892-POOKP  Female              0     Yes         No      28   
12    8091-TTVAX    Male              0     Yes         No      58   
21    1680-VDCWW    Male              0     Yes         No      12   
23    3638-WEABW  Female              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
7023  1035-IPQPU  Female              1     Yes         No      63   
7024  7398-LXGYX    Male              0     Yes         No      44   
7028  9281-CEDRU  Female              0     Yes         No      68   
7031  3605-JISKB    Male              1     Yes         No      55   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
8             Yes               Yes     Fiber optic                   No  ...   
12            Yes               Yes     Fiber optic                   No  ...   
21            Yes                No              No  No internet service  ...   
23            Yes               Yes             DSL                   No  ...   
...           ...               ...             ...                  ...  ...   
7023          Yes               Yes     Fiber optic                   No  ...   
7024          Yes               Yes     Fiber optic                  Yes  ...   
7028          Yes                No             DSL                   No  ...   
7031          Yes               Yes             DSL                  Yes  ...   
7041          Yes               Yes     Fiber optic                   No  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
8                     Yes                  Yes                  Yes   
12                    Yes                   No                  Yes   
21    No internet service  No internet service  No internet service   
23                     No                  Yes                   No   
...                   ...                  ...                  ...   
7023                  Yes                   No                  Yes   
7024                  Yes                   No                   No   
7028                   No                  Yes                  Yes   
7031                   No                   No                   No   
7041                   No                   No                   No   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
8                     Yes  Month-to-month              Yes   
12                    Yes        One year               No   
21    No internet service        One year               No   
23                     No        Two year              Yes   
...                   ...             ...              ...   
7023                  Yes  Month-to-month              Yes   
7024                   No  Month-to-month              Yes   
7028                   No        Two year               No   
7031                   No        One year               No   
7041                   No  Month-to-month              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
8              Electronic check         104.80       3046.05   Yes  
12      Credit card (automatic)         100.35        5681.1    No  
21    Bank transfer (automatic)          19.80        202.25    No  
23      Credit card (automatic)          59.90        3505.1    No  
...                         ...            ...           ...   ...  
7023           Electronic check         103.50        6479.4    No  
7024    Credit card (automatic)          8

In [15]:
jawaban = df_condition2['MonthlyCharges'].max()

In [16]:
jawaban

118.6

## Soal nomor 4

In [17]:
df_condition3 = df_tugas[(df_tugas['Partner'] == 'No') & (df_tugas['Dependents'] == 'Yes')]

In [18]:
df_condition3

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
6     1452-KIOVK    Male              0      No        Yes      22   
9     6388-TABGU    Male              0      No        Yes      62   
17    9959-WOFKT    Male              0      No        Yes      71   
29    8773-HHUOZ  Female              0      No        Yes      17   
74    5630-AHZIL  Female              0      No        Yes       3   
...          ...     ...            ...     ...        ...     ...   
6946  9013-AQORL  Female              0      No        Yes      48   
6966  6598-KELSS    Male              0      No        Yes      50   
6979  5351-QESIO    Male              0      No        Yes       1   
6984  6447-GORXK    Male              0      No        Yes      31   
6995  7164-BPTUT    Male              0      No        Yes      57   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
6             Yes               Yes     Fiber optic             No  ...   
9             Yes                No             DSL            Yes  ...   
17            Yes               Yes     Fiber optic            Yes  ...   
29            Yes                No             DSL             No  ...   
74            Yes                No             DSL            Yes  ...   
...           ...               ...             ...            ...  ...   
6946           No  No phone service             DSL             No  ...   
6966           No  No phone service             DSL            Yes  ...   
6979           No  No phone service             DSL             No  ...   
6984           No  No phone service             DSL             No  ...   
6995          Yes               Yes             DSL            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
6                  No          No         Yes              No  Month-to-month   
9                  No          No          No              No        One year   
17                Yes          No         Yes             Yes        Two year   
29                 No          No         Yes             Yes  Month-to-month   
74                 No         Yes          No             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
6946              Yes          No          No             Yes  Month-to-month   
6966               No          No         Yes              No        One year   
6979               No          No          No              No  Month-to-month   
6984               No          No          No              No  Month-to-month   
6995              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
6                 Yes    Credit card (automatic)          89.10        1949.4   
9                  No  Bank transfer (automatic)          56.15       3487.95   
17                 No  Bank transfer (automatic)         106.70       7382.25   
29                Yes               Mailed check          64.70        1093.1   
74                Yes  Bank transfer (automatic)          64.50         177.4   
...               ...                        ...            ...           ...   
6946               No    Credit card (automatic)          45.55       2108.35   
6966              Yes  Bank transfer (automatic)          43.05       2208.05   
6979               No               Mailed check          24.20          24.2   
6984               No    Credit card (automatic)          25.00         789.2   
6995               No               Mailed check          89.55       5012.35   

     Churn  
6       No  
9       No  
17      No  
29     Yes  
74      No  
...    ...  
6946    No  
6966    No  
6979    No  
6984    No  
6995    No  

[361 rows x 21 columns]

In [19]:
df_condition3['customerID'].nunique()

361

## Soal nomor 5

In [29]:
df['TotalCharges'] = df['TotalCharges'].apply(pd.to_numeric, errors='coerce').fillna(0.0)

In [31]:
df['TotalCharges'].astype(float).astype(int)

0         29
1       1889
2        108
3       1840
4        151
        ... 
7038    1990
7039    7362
7040     346
7041     306
7042    6844
Name: TotalCharges, Length: 7043, dtype: int32

In [34]:
df_condition4 = df[(df['SeniorCitizen'] == 0 ) & (df['PhoneService'] == 'Yes') & (df['InternetService'] == 'Fiber optic')]

In [35]:
df_condition4

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
6     1452-KIOVK    Male              0      No        Yes      22   
8     7892-POOKP  Female              0     Yes         No      28   
12    8091-TTVAX    Male              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
7033  9767-FFLEM    Male              0      No         No      38   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
4             Yes            No     Fiber optic             No  ...   
5             Yes           Yes     Fiber optic             No  ...   
6             Yes           Yes     Fiber optic             No  ...   
8             Yes           Yes     Fiber optic             No  ...   
12            Yes           Yes     Fiber optic             No  ...   
...           ...           ...             ...            ...  ...   
7033          Yes            No     Fiber optic             No  ...   
7034          Yes           Yes     Fiber optic            Yes  ...   
7035          Yes            No     Fiber optic             No  ...   
7039          Yes           Yes     Fiber optic             No  ...   
7042          Yes            No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
4                  No          No          No              No  Month-to-month   
5                 Yes          No         Yes             Yes  Month-to-month   
6                  No          No         Yes              No  Month-to-month   
8                 Yes         Yes         Yes             Yes  Month-to-month   
12                Yes          No         Yes             Yes        One year   
...               ...         ...         ...             ...             ...   
7033               No          No          No              No  Month-to-month   
7034              Yes          No         Yes              No  Month-to-month   
7035               No          No         Yes              No  Month-to-month   
7039              Yes          No         Yes             Yes        One year   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
4                 Yes           Electronic check          70.70        151.65   
5                 Yes           Electronic check          99.65        820.50   
6                 Yes    Credit card (automatic)          89.10       1949.40   
8                 Yes           Electronic check         104.80       3046.05   
12                 No    Credit card (automatic)         100.35       5681.10   
...               ...                        ...            ...           ...   
7033              Yes    Credit card (automatic)          69.50       2625.25   
7034              Yes    Credit card (automatic)         102.95       6886.25   
7035              Yes  Bank transfer (automatic)          78.70       1495.10   
7039              Yes    Credit card (automatic)         103.20       7362.90   
7042              Yes  Bank transfer (automatic)         105.65       6844.50   

      Churn  
4       Yes  
5       Yes  
6        No  
8       Yes  
12       No  
...     ...  
7033     No  
7034    Yes  
7035     No  
7039     No  
7042     No  

[2265 rows x 21 columns]

In [37]:
df_condition4['TotalCharges'].mean()

3226.4927373068463

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 23.303353658536587

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercata?

#### Jawaban: 118.6

#### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 361

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 3226.4927373068463